Mathlib/RingTheory/PowerSeries/WeierstrassPreparation.lean

```diff
@@ -0,0 +1,617 @@
+/-
+Copyright (c) 2025 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.RingTheory.LocalRing.ResidueField.Basic
+import Mathlib.RingTheory.Polynomial.Eisenstein.Distinguished
+import Mathlib.RingTheory.PowerSeries.CoeffMulMem
+import Mathlib.RingTheory.PowerSeries.Inverse
+import Mathlib.RingTheory.PowerSeries.Trunc
+
+/-!
+
+# Weierstrass preparation theorem for power series over a complete local ring
+
+In this file we define Weierstrass division, Weierstrass factorization, and prove
+Weierstrass preparation theorem.
+
+We assume that a ring is adic complete with respect to some ideal.
+If such ideal is a maximal ideal, then by `isLocalRing_of_isAdicComplete_maximal`,
+such ring has only on maximal ideal, and hence such ring is a complete local ring.
+
+## Main definitions
+
+- `PowerSeries.IsWeierstrassDivisionAt f g q r I`: a `Prop` which asserts that a power series
```

Comment:
Can you mention `f` in the description of `PowerSeries.IsWeierstrassDivisionAt`?

Reply:
Sure.